### PR TITLE
Add tests for the automated archiver, and automate the running of those tests

### PR DESCRIPTION
**Pull request by mgwalker**
I am coming late to this party, but I really should have written tests in the first place. Kudos to @geekygirlsarah for asking the obvious question of how to test it, to which the answer was very unsatisfyingly, "in production." Now there are tests so it can be tested.

### CODE REVIEW

File: .github/workflows/archive_closed_jobs.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 const _fs = require("fs/promises");
-const _path = require("path");
+const path = require("path");
```
Comment (mgwalker) [inline .github/workflows/archive_closed_jobs.js:2]: The `path` library as used here doesn't have any side effects, so we don't actually need to mock it.

File: .github/workflows/archive_closed_jobs.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 // Checks if a path exists. Can be a directory or a file. Not STRICTLY a check
 // that the path exists, but rather than the calling user has permission to
 // access it. If it doesn't exist, the user won't have access. Since in this
 // context, our "user" has access to all the paths that we might be checking,
 // we can say the two are basically the same thing.
-const exists = async (path) => {
+const exists = async (pathToCheck, { fs }) => {
   try {
-    await _fs.access(path);
+    await fs.access(pathToCheck);
```
Comment (mgwalker) [inline .github/workflows/archive_closed_jobs.js:11]: The whole reason `fs` was to be injected into the `runArchive` method was so we could mock it, but then the mock wasn't being passed along. This updates `exists` to receive and use the mock.

File: .github/workflows/archive_closed_jobs.js [hunk 3]
```diff
@@ -56,7 +55,7 @@ const runArchiver = async ({ core, fs = _fs, path = _path }) => {
       const archiveBase = path.join("archive", path.basename(pathname));
       let archivePath = archiveBase;
       let offset = 0;
-      while (await exists(archivePath)) {
+      while (await exists(archivePath, { fs })) {
```
Comment (mgwalker) [inline .github/workflows/archive_closed_jobs.js:58]: 🎵 Pass it in! Pass it in! 🎵

**Review by geekygirlsarah — APPROVED** [review]
Sweet, looks good and GH Actions seems happy with it. Woo hoo!

**Pull request merged**